### DISCUSSION TITLE
MSVC build with rustflags = ["-C", "target-feature=+crt-static"]

### DISCUSSION THREAD

**Discussion by kurbatxx** [fltk-rs/fltk-rs#1151] [General]
## Desktop info
 - OS: [e.g. Windows 10 x64]

Checking whether this env can build fltk-rs..
This is testing an MSVC environment on Windows
Found suitable Rust version!
Found working git executable!
Found working CMake executable!
Found working Ninja executable!
Found a C++ compiler!
Found C++ compiler supporting C++11!
Found library: comdlg32!
Found library: ole32!
Found library: ws2_32!
Found library: odbc32!
Found library: winspool!
Found library: comctl32!
Found library: gdiplus!
Found library: uuid!
Found library: user32!
Found library: gdi32!
Found library: shell32!
Found library: opengl32!
Found library: kernel32!
Found library: oleaut32!
Found library: advapi32!
Found library: glu32!

## Additional info
_.cargo/config.toml_
```
[target.x86_64-pc-windows-msvc]
rustflags = ["-C", `"target-feature=+crt-static"]
```
```
[dependencies]
fltk = { version = "^1.2"}
```

## Build failures
error: linking with `link.exe` failed: exit code: 1120

= note: "C:\\DEV\\VS_2019\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\link.exe" "/NOLOGO" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.0.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.1.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.2.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.3.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.4.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.5.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.6.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.7.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.hello_fltk.c9f021d2-cgu.8.rcgu.o" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.2vpaxplnm23b2iqt.rcgu.o" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\build" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\build\\Release" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\lib" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\lib64" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\lib\\Release" "/LIBPATH:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\build\\fltk-sys-2c3f1920067dcbb5\\out\\lib64\\Release" "/LIBPATH:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libfltk-5e92fda8460791dd.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libttf_parser-e49c832dff57aedc.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libcrossbeam_channel-efbcfdea60694c58.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libcrossbeam_utils-6ab3701a41c11bae.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\liblazy_static-274044cadaed1c64.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libcfg_if-e9ba1d238e955bdd.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libbitflags-6f19ec4e4b9f9c3b.rlib" "C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\libfltk_sys-73069486e5bcca5c.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd-ca7b0c28ec762872.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libpanic_unwind-1b050a71ed5c4477.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libstd_detect-ba9f9c006950f110.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_demangle-4e228cbffbe5e3cd.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libhashbrown-d3d7c65121bb0d35.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_alloc-2cedaf2947cb8622.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libunwind-7d6cff0c7b8f0c2b.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcfg_if-63bdfcda4a65748c.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liblibc-c331c9d260094b22.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\liballoc-34d0a2dd4a5dbc91.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\librustc_std_workspace_core-e7b8421abede5598.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcore-08b052fa5e861ac2.rlib" "C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib\\libcompiler_builtins-18761c3bc8f2e6ea.rlib" "cfltk.lib" "kernel32.lib" "ws2_32.lib" "bcrypt.lib" "advapi32.lib" "userenv.lib" "kernel32.lib" "msvcrt.lib" "/NXCOMPAT" "/LIBPATH:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "/OUT:C:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.exe" 
"/SUBSYSTEM:windows" "/ENTRY:mainCRTStartup" "/OPT:REF,ICF" "/DEBUG" "/NATVIS:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\intrinsic.natvis" "/NATVIS:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\liballoc.natvis" "/NATVIS:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libcore.natvis" "/NATVIS:C:\\DEV\\Rust\\.rustup\\toolchains\\1.59-x86_64-pc-windows-msvc\\lib\\rustlib\\etc\\libstd.natvis"
  = note: Non-UTF-8 output: libfltk-5e92fda8460791dd.rlib(fltk-5e92fda8460791dd.fltk.59488003-cgu.5.rcgu.o) : error LNK2019: \xe1\xe1\xeb\xab\xaa\xa0 \xad\xa0 \xad\xa5\xe0\xa0\xa7\xe0\xa5\xe8\xa5\xad\xad\xeb\xa9 \xa2\xad\xa5\xe8\xad\xa8\xa9 \xe1\xa8\xac\xa2\xae\xab ShowWindow \xa2 \xe4\xe3\xad\xaa\xe6\xa8\xa8 _ZN4fltk6window12DoubleWindow13platform_show17h78a43848c556f869E.\r\nC:\\Users\\Kurbat\\Desktop\\hello-fltk\\target\\release\\deps\\hello_fltk.exe : fatal error LNK1120: \xad\xa5\xe0\xa0\xa7\xe0\xa5\xe8\xa5\xad\xad\xeb\xe5 \xa2\xad\xa5\xe8\xad\xa8\xe5 \xed\xab\xa5\xac\xa5\xad\xe2\xae\xa2: 1\r\n

**Comment by MoAlyousef** [comment]
I'm not sure that crt-static is attainable with the msvc target for fltk. You can try with the gnu target.
